packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.tsx

```diff
@@ -79,7 +71,28 @@ function formatValue(key: string, value: any) {
   } else if (Array.isArray(parsed) && shouldDisplayAsStringList(key)) {
     content = stringListMarkup(parsed);
   } else {
-    content = _jsonMarkup(parsed);
+    const shouldJsonTreeExpand = Object.keys(parsed).length <= 10;
+
+    content = (
+      <JsonView
+        data={parsed}
+        shouldInitiallyExpand={shouldJsonTreeExpand ? allExpanded : collapseAllNested}
+        style={{
+          ...defaultStyles,
+          container: 'json-markup',
+          label: 'json-markup-key',
+          stringValue: 'json-markup-string',
+          numberValue: 'json-markup-number',
+          booleanValue: 'json-markup-bool',
+          nullValue: 'json-markup-null',
+          undefinedValue: 'json-markup-undefined',
+          expander: 'json-markup-expander',
+          basicChildStyle: 'json-markup-child',
```

Comment:
This doesn't look like a desirable outcome:

<img width="1552" alt="image" src="https://github.com/jaegertracing/jaeger-ui/assets/89186/f51b2fae-fa33-4cc3-9a33-58f63890ce0b">

Reply:
please elaborate

Reply:
I made https://github.com/jaegertracing/jaeger-ui/pull/2168